lib/os/heap-validate.c

```diff
@@ -386,3 +386,66 @@ void sys_heap_print_info(struct sys_heap *heap, bool dump_chunks)
 {
 	heap_print_info(heap->heap, dump_chunks);
 }
+
+#ifdef CONFIG_SYS_HEAP_RUNTIME_STATS
+
+int heap_runtime_stats_get(struct sys_heap *heap,
+		struct sys_heap_runtime_stats *stats)
+{
+	size_t free_bytes = 0;
+	struct z_heap *h;
+	chunkid_t c = 0;
+
+	if ((heap == NULL) || (stats == NULL)) {
+		return -EINVAL;
+	}
+
+	h = heap->heap;
+	stats->free_bytes = 0;
+	stats->allocated_bytes = 0;
+	stats->max_avail_bytes = 0;
+
+	/* Iterate through all chunks */
```

Comment:
Do we really want a O(N) implementation here?  Most of the time people just want to know the sum of all the sizes passed to successful allocations.  That can be tracked with a simple increment/decrement hook at allocation/free time.